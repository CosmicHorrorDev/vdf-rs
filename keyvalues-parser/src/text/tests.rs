use std::{error::Error, fs, path::Path};

use crate::Vdf;

type BoxedResult<T> = Result<T, Box<dyn Error>>;

fn read_asset_file(file_name: &str) -> BoxedResult<String> {
    let val = fs::read_to_string(Path::new("tests").join("assets").join(file_name))?;
    Ok(val)
}

fn snapshot_test_parse_and_render(file_name: &str) -> BoxedResult<()> {
    let vdf_text = read_asset_file(file_name)?;
    let vdf = Vdf::parse(&vdf_text)?;
    insta::assert_ron_snapshot!(vdf);

    let rendered = vdf.to_string();
    insta::assert_snapshot!(rendered);

    Ok(())
}

macro_rules! parse_render_test {
    ($func_name:ident, $file_name: literal) => {
        #[test]
        fn $func_name() -> BoxedResult<()> {
            snapshot_test_parse_and_render($file_name)
        }
    };
}

parse_render_test!(basic, "basic.vdf");
parse_render_test!(app_manifest, "app_manifest.vdf");
parse_render_test!(comments, "comments.vdf");
parse_render_test!(unquoted_strings, "unquoted_strings.vdf");
parse_render_test!(special_charaters, "special_characters.vdf");
parse_render_test!(app_info, "app_info.vdf");
