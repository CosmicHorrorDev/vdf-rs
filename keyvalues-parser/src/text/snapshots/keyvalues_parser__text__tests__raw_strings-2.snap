---
source: keyvalues-parser/src/text/tests.rs
assertion_line: 33
expression: buf

---
"Some\text\here"
{
	"more\text\here"	"and\here"
}

